packages/oui-file/README.md

```diff
@@ -0,0 +1,117 @@
+```
+
+### Placeholder
+
+```html:preview
+<oui-file model="$ctrl.modelPlaceholder" placeholder="Placeholder example"></oui-file>
+```
+
+### Disabled
+
+```html:preview
+<oui-file model="$ctrl.modelDisabled" disabled></oui-file>
+```
+
+### File restriction
+
+<oui-message type="info" dismissable="false">`maxsize` support form validators</oui-message>
+
+```html:preview
+<form novalidate name="fileForm">
+    <oui-field label="Upload file" help-text="image/jpeg, image/png, image/gif (max size 150 KB)" size="xl">
+        <oui-file name="fileUpload" model="$ctrl.modelField" 
+            accept="image/jpeg,image/png,image/gif" 
+            maxsize="150000" 
+            required>
+        </oui-file>
+    </oui-field>
+    <p><strong><code class="oui-doc-codespan">fileForm.$valid</code> value:</strong> {{fileForm.$valid | json}}</p>
+    <p><strong><code class="oui-doc-codespan">fileForm.$submitted</code> value:</strong> {{fileForm.$submitted | json}}</p>
+    <p><strong><code class="oui-doc-codespan">fileForm["fileUpload"].$error</code> value:</strong> {{fileForm["fileUpload"].$error | json}}</p>
+    <oui-button type="submit">Submit</oui-button>
+</form>
+```
+
+## Multiple files
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelMultiple" maxsize="150000" multiple></oui-file>
+```
+
+### With preview
+
+<oui-message type="info" dismissable="false">Preview works only with `image/*` files.</oui-message>
+
+```html:preview
+<oui-file model="$ctrl.modelMultiplePreview" maxsize="150000" multiple preview></oui-file>
+```
+
+## Drag & Drop area
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelDroparea" maxsize="150000" droparea></oui-file>
+```
+
+### With preview
+
+<oui-message type="info" dismissable="false">Preview works only with `image/*` files.</oui-message>
+
+```html:preview
+<oui-file model="$ctrl.modelDropareaPreview" maxsize="150000" droparea preview></oui-file>
+```
+
+## API
+
+| Attribute         | Type              | Binding   | One-time binding  | Values            | Default   | Description
+| ----              | ----              | ----      | ----              | ----              | ----      | ----
+| `model`           | array&lt;file&gt; | =         | no                | n/a               | n/a       | model bound to component
+| `id`              | string            | @?        | yes               | n/a               | n/a       | id attribute of form input
+| `name`            | string            | @?        | yes               | n/a               | n/a       | name attribute of form input
+| `placeholder`     | string            | @?        | yes               | n/a               | n/a       | placeholder text
+| `accept`          | string            | @?        | yes               | n/a               | n/a       | accept attribute of file input
+| `maxsize`         | number            | <?        | no                | n/a               | n/a       | maximum file size in byte
```

Comment:
`on-select` but `maxsize` ? (`on-select` is separated by an hyphen but `maxsize` is all attached)

Reply:
default max size ? Users could try to upload 1 Tb files and think that their browser is frozen because of us

Reply:
wouldn't it make sense to be able to select a unit too ?

Reply:
Like we can do with the current input file ?

`on-select` -> Keep our way on naming event
`maxsize` -> same as `maxlength` attribute